# awesome-genome-visualization

A list of interesting genome visualizers, genome browsers, or genome-browser-like implementations

## Genome browsers

* [Alamut](https://www.interactive-biosoftware.com/alamut-visual/) (commercial)
* [ATCC Genome Portal](https://genomes.atcc.org) (commercial)
* [CLC Genomics workbench](https://digitalinsights.qiagen.com/news/blog/discovery/structural-variant-detection-using-clc-genomics-workbench/) (commercial)
* Ensembl genome browser
* GBrowse
* GBrowse 2
* [GenomeMaps](http://www.genomemaps.org/) (used on ICGC)
* [Golden Helix](http://goldenhelix.com/GenomeBrowse/) (commercial)
* IGB
* IGV (java)
* IGV.js
* [JBrowse](http://jbrowse.org/)
* [MacVector](https://macvector.com/) (commerical)
* NCBI Genome Data Viewer
* [Persephone](https://persephonesoft.com/) (commercial)
* [Trackster](https://galaxyproject.org/learn/visualization/) (in Galaxy)
* UCSC genome browser
* [WashU epigenomics browser](https://epigenomegateway.wustl.edu/) [source](https://github.com/lidaof/eg-react)

## Other

* [ApE](https://jorgensen.biology.utah.edu/wayned/ape/) (a plasmid editor)
* [Apollo](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC151184/) (old desktop version)
* [Apollo aka WebApollo](https://genomearchitect.readthedocs.io/en/latest/)
* [2020 Ensembl genome browser](http://2020.ensembl.org/)
* [Artemis comparison tool (ACT)](https://www.sanger.ac.uk/science/tools/artemis-comparison-tool-act)
* [BasePlayer](https://baseplayer.fi/)
* [Biodalliance](http://www.biodalliance.org/)
* [BrowseVCF](https://github.com/BSGOxford/BrowseVCF)
* [Chip Monk](http://www.bioinformatics.babraham.ac.uk/projects/chipmonk/)
* [Chipster](https://chipster.csc.fi/)
* [cisGenome Browser](http://www-personal.umich.edu/~jianghui/browser/screenshots.html)
* [CNVnator](http://www.abyzovlab.org/cnvnatorapplet)
* [Consed](http://www.phrap.org/consed/consed.html)
* [CuteVCF](https://github.com/labsquare/CuteVCF)
* [DNASkittle](https://www.dnaskittle.com/)
* [EaSeq](https://easeq.net/screenshots/)
* [Epilogos](http://compbio.mit.edu/epilogos/)
* [Epiviz](https://www.biorxiv.org/content/10.1101/865295v1.full.pdf)
* [GenomeRibbon](http://genomeribbon.com/)
* [GenomeSpy](https://genomespy.app/examples/TCGA/BRCA/) [another demo](https://genomespy.app/examples/GWAS/OCAC-HGSOC/)
* [Genoverse](https://github.com/wtsi-web/Genoverse)
* [Gnomad browser](https://github.com/macarthur-lab/gnomadjs)
* [Gremlin](http://compbio-research.cs.brown.edu/software/Gremlin/GremlinOverview.pdf)
* [GWAS catalog browser](https://www.ebi.ac.uk/gwas/variants/rs1558902) at EBI
* [Hawkeye](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2007-8-3-r34)
* [Human genome dating](https://human.genome.dating/region/chr1_13250000) (made with vega/d3)
* [HiC browser](http://promoter.bx.psu.edu/hi-c/view.php) - UCSC-based
* [HiGlass](https://higlass.io)
* [IslandPlot](https://bl.ocks.org/lairdm/c6c235dbfa6e6ee61565) (D3, SVG) ([source](https://github.com/lairdm/islandplot))
* [Juicebox](https://github.com/aidenlab/Juicebox)
* [Locuszoom](http://locuszoom.org/)
* [LookSeq](https://genome.cshlp.org/content/early/2009/08/13/gr.093443.109)
* [MagicViewer](http://bioinformatics.zj.cn/magicviewer/)
* [NGB](https://github.com/epam/NGB)
* [Parasight](https://baileylab.brown.edu/parasight/galframe.html)
* [Pileup.js](https://github.com/hammerlab/pileup.js/)
* [ProteinPaint](https://pecan.stjude.cloud/proteinpaint)
* [PureScript genetics browser](https://github.com/chfi/purescript-genetics-browser)
* [RNASeqBrowser](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-015-1346-2) UCSC-based
* [SEQing](https://github.com/malewins/SEQing)
* [Savant](http://bioinformatics-ca.github.io/savant_genome_browser_lab_2015/)
* [SNPitty](https://jmd.amjpathol.org/article/S1525-1578(17)30166-6/fulltext)
* [Staden](https://sourceforge.net/projects/staden/)
* [SplitThreader](http://splitthreader.com/)
* [TASUKE](https://ricegenome.dna.affrc.go.jp/) - from RAP-DB
* [Tablet](https://ics.hutton.ac.uk/tablet/)
* [UGENE](http://ugene.net/)
* [Valis browser](https://valis.bio/) - used on encodeproject.org (uses webGL)
* [Varsome](https://varsome.com/security-validation/?next=/variant/hg19/NM_000088.3(COL1A1):c.658C%3ET#community-contributions)
* [Vials](http://vials.io/info/)
* [WebAsm demo browser](https://shk656461.github.io/index.html)
* [UCSC Xena](https://xena.ucsc.edu/)
* [Zenbu](http://fantom.gsc.riken.jp/zenbu/)

## Heatmaps


* [VIVA](https://www.biorxiv.org/content/10.1101/589879v1.full.pdf)


## Genome map, or comparative viewers

* [Aequatus](https://github.com/TGAC/Aequatus)
* [Biodalliance comparative demo](http://biodalliance.org/dev/test-comparative.html)
* [Chromatiblock](https://www.biorxiv.org/content/10.1101/800920v2.full.pdf)
* [CMap](http://gmod.org/wiki/CMap)
* [CoGe](https://genomevolution.org/coge/)
* [CVit](https://sourceforge.net/projects/cvit/)
* [Cvit.js](https://github.com/LegumeFederation/cvitjs)
* [CrossBrowse](https://github.com/shenkers/CrossBrowse) [pub](https://www.biorxiv.org/content/10.1101/272880v1.full)
* [GIVE](https://zhong-lab-ucsd.github.io/GIVE_homepage/)
* [GBrowse_syn](http://gmod.org/wiki/GBrowse_syn)
* [Genome context viewer](https://www.legumefederation.org/en/blog/2019/06/25/genome-context-viewer-gcv-test-drive/)
* [MCScan (python version)](https://github.com/tanghaibao/jcvi/wiki/MCscan-(Python-version))
* [MCScanX](http://chibba.pgml.uga.edu/mcscan2/)
* [MizBee](https://ieeexplore.ieee.org/document/5290692)
* [Multiple genome viewer](http://www.informatics.jax.org/mgv/) (jax)
* [mGSV](https://github.com/kashmatic/mGSV)
* [progressiveMauve viewer](http://darlinglab.org/mauve/user-guide/viewer.html)
* [Synteny browser](https://github.com/TheJacksonLaboratory/syntenybrowser) (jax)
* [Synima](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5697234/)
* [VISTA browser](http://pipeline.lbl.gov/cgi-bin/gateway2)

## Multiple sequence alignment viewer/Logo

* [BioJS/MSA](https://msa.biojs.net/) also see [msa/R](https://github.com/zachcp/msaR)
* [Jalview](https://www.jalview.org/)
* [Wasabi](http://wasabiapp.org/)
* [AlignmentViewer](https://alignmentviewer.org/)
* [MView](https://www.ebi.ac.uk/Tools/msa/mview/)
* [TeXShade](http://mirrors.ibiblio.org/CTAN/macros/latex/contrib/texshade/texshade.pdf)
* [ESPript](http://espript.ibcp.fr/ESPript/ESPript/)
* [LogoJS](https://logojs.wenglab.org/app/gallery/)


## Static visualizations

* [aCNVViewer](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0189334)
* [AliTV](https://zenodo.org/record/32014)
* [BAMScale](https://github.com/ncbi/BAMscale)
* [CNView](https://github.com/RCollins13/CNView)
* [Chimeraviz](https://github.com/stianlagstad/chimeraviz)
* [ChIA-Pipe](https://www.biorxiv.org/content/10.1101/506683v1.full.pdf)
* [ClustersPloter](https://github.com/orangeSi/ClustersPloter)
* [CNVPlot](https://github.com/dantaki/CNVplot)
* [DNAPlotLib](https://github.com/VoigtLab/dnaplotlib)
* [ExonIntron](http://wormweb.org/exonintron) from WormBase
* [FeatureViewer](https://github.com/calipho-sib/feature-viewer)
* [ggbio](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2012-13-8-r77)
* [GGgenes](https://github.com/wilkox/gggenes)
* [GGsashimi](https://github.com/guigolab/ggsashimi)
* [genoPlotR](http://genoplotr.r-forge.r-project.org/)
* [Geneviz](https://jrderuiter.github.io/geneviz/usage.html)
* [Genome STRiP](http://software.broadinstitute.org/software/genomestrip/)
* [GenomeTools](http://genometools.org/)
* [GenomeView](https://github.com/nspies/genomeview)
* [GSDraw](https://wheat.pw.usda.gov/piece/GSDraw.php)
* [Gviz](https://bioconductor.org/packages/release/bioc/html/Gviz.html)
* [Hagfish](https://github.com/mfiers/hagfish/wiki/Plots)
* [HiCPlotter](https://github.com/kcakdemir/HiCPlotter)
* [JVarKit](http://lindenb.github.io/jvarkit/) examples http://lindenb.github.io/jvarkit/BamToSVG.html
* [Lollipops](https://github.com/pbnjay/lollipops)
* [Methplotlib](https://www.biorxiv.org/content/10.1101/826107v1.full.pdf)
* [RACER](https://oliviasabik.github.io/RACERweb/articles/IntroToRACER.html)
* [RepViz](https://bmcresnotes.biomedcentral.com/articles/10.1186/s13104-019-4473-z)
* [Samplot](https://github.com/ryanlayer/samplot)
* [SashimiPlot](https://miso.readthedocs.io/en/fastmiso/sashimi.html)
* [SparK](https://www.biorxiv.org/content/10.1101/845529v1.full.pdf)
* [SNPduo](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006711)
* [Spliceclust](https://github.com/pkimes/spliceclust)
* [Sushi](https://www.bioconductor.org/packages/release/bioc/vignettes/Sushi/inst/doc/Sushi.pdf) (R package)
* [pyGenomeTracks](https://github.com/deeptools/pyGenomeTracks)
* [svist4get](https://link.springer.com/article/10.1186/s12859-019-2706-8)
* [svviz2](https://github.com/nspies/svviz2)
* [svviz](https://github.com/svviz/svviz)
* [TnT](https://tnt.marlin.pub/articles/introduction)
* [trackViewer](https://bioconductor.org/packages/release/bioc/vignettes/trackViewer/inst/doc/trackViewer.html) (R)


## Circular

* [Anvio](https://peerj.com/articles/1319/)
* [Circos](http://circos.ca/)
* [CircosJS](https://github.com/nicgirault/circosJS)
* [OGDRAW](https://www.biorxiv.org/content/10.1101/545509v1.full.pdf)


## Dotplot

* [D-GENIES](https://github.com/genotoul-bioinfo/dgenies)
* [Delly-maze](https://github.com/dellytools/maze)
* [Dot](https://github.com/dnanexus/dot)
* [Dotplot](https://github.com/iLambda/dotplot)
* [Gepard](http://cube.univie.ac.at/gepard)
* [Jdot](https://github.com/LyonsLab/jdot)
* LAST (package contains a dotplotter)
* [Minidot](https://github.com/thackl/minidot)
* Mummer (contains a GNUplot script for dotplotting)
* [Redottable](https://github.com/s-andrews/redotable)
* [syntenyPlotByR](https://github.com/shingocat/syntenyPlotByR/blob/master/out.png)

## Karyotype

* [chromoMap](https://www.biorxiv.org/content/10.1101/605600v1.full.pdf)
* [Ideogram](https://github.com/RCollins13/HumanIdiogramLibrary) HumanIdeogramLibrary
* [eweitz/Ideogram](https://github.com/eweitz/ideogram)
* [KaryotypeSVG](https://github.com/andreasprlic/karyotypeSVG)
* [rnaseqview](https://github.com/NCBI-Hackathons/rnaseqview)
* [karyoploteR](https://github.com/bernatgel/karyoploteR) (R package)


## Command line

* [ASCIIGenome](https://github.com/dariober/ASCIIGenome)
* [Alan](https://github.com/mpdunne/alan)
* [Alv](https://github.com/arvestad/alv)
* [Bamcov](https://github.com/fbreitwieser/bamcov)
* [bioSyntax](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2315-y)
* [BLAST+](ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/) <-- reading -outfmt 0 pairwise output is essentially a textual/command line bioinformatics tool
* [Hapviz](https://github.com/ekg/hapviz)
* [Sam2pairwise](https://github.com/mlafave/sam2pairwise)
* [Samtools tview](http://www.htslib.org/)
* [VizAln](https://github.com/tfwillems/HipSTR#alignment-visualization) from HipSTR

Given some of the above definitions, human headable text files like BLAST output e.g. -outfmt 0 with the textual

## Graph genomes

* [Bandage](https://github.com/rrwick/Bandage/)
* [gGnomes.js](https://github.com/mskilab/gGnome.js)
* [MoMi-G](https://github.com/MoMI-G/MoMI-G/)
* [sequence tube map](https://github.com/vgteam/sequenceTubeMap)


# Other lists

* [Awesome Bioinformatics](https://github.com/danielecook/Awesome-Bioinformatics)

Send in PRs for more stuff!
